resource/cmake/CommonFunctions.cmake

```diff
@@ -309,20 +213,23 @@ macro(rtigw_init_globals)
 
     # Set common variables
     set(COMMON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/common")
-    set(JSON_PARSER_WRAPPER_DIR "${COMMON_DIR}/json_parser")
-    set(DDS_COMMON_DIR "${COMMON_DIR}/dds_specific")
-    set(TRANSFORMATION_COMMON_DIR "${COMMON_DIR}/transformation")
-    set(THIRD_PARTY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
-    set(LIBMODBUS_DIR "${THIRD_PARTY_DIR}/libmodbus")
-    set(PAHO_MQTT_C_DIR "${THIRD_PARTY_DIR}/paho.mqtt.c")
     set(RESOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/resource")
-    set(UTILS_COMMON_DIR "${COMMON_DIR}/utils")
-    set(JSON_PARSER_DIR "${THIRD_PARTY_DIR}/json_parser")
+    set(THIRD_PARTY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/third-party")
     set(ADAPTERS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/adapters")
     set(PROCESSORS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/processors")
     set(TRANSFORMATIONS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/transformations")
     set(EXAMPLES_DIR "${CMAKE_CURRENT_SOURCE_DIR}/examples")
 
+    set(LIBMODBUS_DIR "${THIRD_PARTY_DIR}/libmodbus")
+    set(PAHO_MQTT_C_DIR "${THIRD_PARTY_DIR}/paho.mqtt.c")
+    set(JSON_PARSER_DIR "${THIRD_PARTY_DIR}/json_parser")
+    set(UTILS_COMMON_DIR "${COMMON_DIR}/utils")
+    set(JSON_PARSER_WRAPPER_DIR "${COMMON_DIR}/json_parser")
+    set(TRANSFORMATION_COMMON_DIR "${COMMON_DIR}/transformation")
+    set(DDS_COMMON_DIR "${COMMON_DIR}/dds_specific")
+    set(CMAKE_COMMON_DIR "${RESOURCE_DIR}/cmake")
+    set(CMAKE_UTILS_MODULES_DIR "${THIRD_PARTY_DIR}/rticonnextdds-cmake-utils/cmake/Modules")
```

Comment:
Let's use a different namespace for these variables to avoid any potential clash with CMake's own variable. E.g. `RTICMAKE_COMMON_DIR` and `RTICMAKE_MODULES_DIR`